client/src/pages/Search/AddressSearch.js

```diff
@@ -20,28 +22,43 @@ export default function AddressSearch({ updateFields }) {
       componentRestrictions: { country: "usa" },
     },
   });
+  async function userAddress({ updateFields }) {
+    try {
+      const coordinates = await getUserCoordinates();
+      const latitude = coordinates.coordinates.latitude;
+      const longitude = coordinates.coordinates.longitude;
+      updateFields({ longitude, latitude });
+    } catch {
+      alert("Please Enable Location Services");
+    }
+  }
 
   return (
-    <TextField
-      inputRef={googlePlacesRef}
-      variant="standard"
-      sx={{ width: "60%", minWidth: "300px" }}
-      label="Address"
-      defaultValue={HUNTER_COLLEGE_ADDRESS}
-      InputProps={{
-        endAdornment: (
-          <InputAdornment position="end" padding={0}>
-            <IconButton
-              onClick={() => {
-                googlePlacesRef.current.value = "";
-              }}
-              sx={{ p: 0.25 }}
-            >
-              <ClearIcon />
-            </IconButton>
-          </InputAdornment>
-        ),
-      }}
-    />
+    <div>
+      <TextField
+        inputRef={googlePlacesRef}
+        variant="standard"
+        sx={{ width: "60%", minWidth: "300px" }}
+        label="Address"
+        defaultValue={HUNTER_COLLEGE_ADDRESS}
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end" padding={0}>
+              <IconButton
+                onClick={() => {
+                  googlePlacesRef.current.value = "";
+                }}
+                sx={{ p: 0.25 }}
+              >
+                <ClearIcon />
+              </IconButton>
+            </InputAdornment>
+          ),
+        }}
+      />
+      <IconButton onClick={() => userAddress({ updateFields })}>
+        <MyLocationIcon />
+      </IconButton>
```

Comment:
This should be an endAdornment or startAdornment. See InputProps of the TextField right above this.